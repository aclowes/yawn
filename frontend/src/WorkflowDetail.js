import React from 'react';
import {Panel, PanelGroup} from 'react-bootstrap';
//import {Link} from 'react-router';

import API from "./API";
import WorkflowDetailGraph from "./WorkflowDetailGraph";
import WorkflowDetailHistory from "./WorkflowDetailHistory";
import WorkflowDetailForm from "./WorkflowDetailForm";

export default class WorkflowDetail extends React.Component {
  constructor(props) {
    super(props);
    this.state = {workflow: null};
  }

  componentDidMount() {
    API.get(`/api/workflows/${this.props.params.id}/`, (payload, error) => {
      this.setState({workflow: payload, error: error});
    })
  }

  render() {
    const workflow = this.state.workflow;
    if (workflow) {
      return (
        <PanelGroup>
          <Panel header="Details">
            <WorkflowDetailForm workflow={workflow}/>
          </Panel>
          <Panel header="Tasks">
            <WorkflowDetailGraph workflow={workflow}/>
          </Panel>
          <Panel header="Runs">
            <WorkflowDetailHistory workflow={workflow}/>
          </Panel>
        </PanelGroup>
      );
    } else {
      return <span>Loading...</span>
    }
  }
}
