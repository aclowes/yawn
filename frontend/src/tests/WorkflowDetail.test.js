import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import WorkflowDetail from '../WorkflowDetail';
import {mockAPI} from './mocks'

it('WorkflowDetail loading', () => {
  mockAPI([null]);
  const component = ReactTestRenderer.create(<WorkflowDetail params={{id: 1}}/>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('WorkflowDetail failure', () => {
  mockAPI();
  const component = ReactTestRenderer.create(<WorkflowDetail params={{id: 1}}/>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('WorkflowDetail success', () => {
  mockAPI([
    {
      "id": 1,
      "name": "Simple Workflow Example",
      "name_id": 1,
      "tasks": [{
        "id": 1,
        "queue": "default",
        "upstream": [],
        "name": "start",
        "command": "echo Starting...",
        "max_retries": 0,
        "timeout": null
      }, {
        "id": 2,
        "queue": "default",
        "upstream": ["start"],
        "name": "task2",
        "command": "echo Working on $MY_OBJECT_ID",
        "max_retries": 0,
        "timeout": null
      }, {
        "id": 3,
        "queue": "default",
        "upstream": ["start"],
        "name": "task3",
        "command": "echo Another busy thing && sleep 20",
        "max_retries": 0,
        "timeout": null
      }, {
        "id": 4,
        "queue": "default",
        "upstream": ["task2", "task3"],
        "name": "done",
        "command": "echo Finished!",
        "max_retries": 0,
        "timeout": null
      }],
      "version": 1,
      "schedule_active": false,
      "schedule": null,
      "next_run": null,
      "parameters": {"MY_OBJECT_ID": "1", "SOME_SETTING": "false"}
    },
    // version list
    {
      "id": 1,
      "current_version": 1,
      "current_version_id": 1,
      "task_count": 4,
      "schedule": null,
      "schedule_active": false,
      "versions": [1],
      "name": "Simple Workflow Example"
    },
    // run history
    [{
      "id": 1,
      "tasks": [{
        "id": 1,
        "name": "start",
        "workflow": {"id": 1, "name": "Simple Workflow Example", "version": 1},
        "status": "succeeded"
      }, {
        "id": 2,
        "name": "task2",
        "workflow": {"id": 1, "name": "Simple Workflow Example", "version": 1},
        "status": "succeeded"
      }, {
        "id": 3,
        "name": "task3",
        "workflow": {"id": 1, "name": "Simple Workflow Example", "version": 1},
        "status": "succeeded"
      }, {
        "id": 4,
        "name": "done",
        "workflow": {"id": 1, "name": "Simple Workflow Example", "version": 1},
        "status": "succeeded"
      }],
      "workflow_id": 1,
      "submitted_time": "2017-01-22T20:53:36.548913Z",
      "scheduled_time": null,
      "status": "succeeded",
      "parameters": {"child": "true", "MY_OBJECT_ID": "1", "SOME_SETTING": "false"},
      "workflow": 1
    }]
  ]);
  const component = ReactTestRenderer.create(<WorkflowDetail params={{id: 1}}/>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
