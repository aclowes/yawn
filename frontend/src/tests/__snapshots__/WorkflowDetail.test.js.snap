// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowDetail failure 1`] = `
<div>
  <div
    className="alert alert-danger"
    role="alert"
  >
    some error message
  </div>
  Loading...
</div>
`;

exports[`WorkflowDetail loading 1`] = `
<div>
  Loading...
</div>
`;

exports[`WorkflowDetail success 1`] = `
<div
  className="panel-group"
>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-heading"
    >
      Details
    </div>
    <div
      className="panel-body"
    >
      <div>
        <dl
          className="dl-horizontal"
        >
          <dt>
            Name
          </dt>
          <dd>
            Simple Workflow Example
          </dd>
          <dt>
            Version
          </dt>
          <dd>
            <ul
              className="pagination pagination-sm"
            >
              <li
                className="active"
                style={undefined}
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="button"
                >
                  1
                </a>
              </li>
            </ul>
          </dd>
          <dt>
            Schedule Active
          </dt>
          <dd>
            False
          </dd>
          <dt>
            Schedule
          </dt>
          <dd />
          <dt>
            Next Run
          </dt>
          <dd>
            
          </dd>
          <dt>
            Parameters
          </dt>
          <dd>
            <pre>
              MY_OBJECT_ID=1
              SOME_SETTING=false
            </pre>
          </dd>
        </dl>
        <button
          className="btn btn-default"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Edit
        </button>
      </div>
    </div>
  </div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-heading"
    >
      Tasks
    </div>
    <div
      className="panel-body"
    >
      <svg />
    </div>
  </div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-heading"
    >
      Runs
    </div>
    <div
      className="panel-body"
    >
      <div>
        <table
          className="table table-bordered table-condensed"
        >
          <thead>
            <tr>
              <th
                className="text-center"
              >
                Task
              </th>
              <th
                className="text-center"
                colSpan={1}
              >
                Run
              </th>
            </tr>
            <tr>
              <th />
              <th
                className="succeeded"
                onBlur={[Function]}
                onClick={null}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
              >
                #
                0
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                start
              </td>
              <td>
                <a
                  onClick={[Function]}
                  style={Object {}}
                >
                  <div
                    onBlur={[Function]}
                    onClick={null}
                    onFocus={[Function]}
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                  >
                    <span
                      className="succeeded glyphicon glyphicon-ok"
                    />
                  </div>
                </a>
              </td>
            </tr>
            <tr>
              <td>
                task2
              </td>
              <td>
                <a
                  onClick={[Function]}
                  style={Object {}}
                >
                  <div
                    onBlur={[Function]}
                    onClick={null}
                    onFocus={[Function]}
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                  >
                    <span
                      className="succeeded glyphicon glyphicon-ok"
                    />
                  </div>
                </a>
              </td>
            </tr>
            <tr>
              <td>
                task3
              </td>
              <td>
                <a
                  onClick={[Function]}
                  style={Object {}}
                >
                  <div
                    onBlur={[Function]}
                    onClick={null}
                    onFocus={[Function]}
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                  >
                    <span
                      className="succeeded glyphicon glyphicon-ok"
                    />
                  </div>
                </a>
              </td>
            </tr>
            <tr>
              <td>
                done
              </td>
              <td>
                <a
                  onClick={[Function]}
                  style={Object {}}
                >
                  <div
                    onBlur={[Function]}
                    onClick={null}
                    onFocus={[Function]}
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                  >
                    <span
                      className="succeeded glyphicon glyphicon-ok"
                    />
                  </div>
                </a>
              </td>
            </tr>
          </tbody>
        </table>
        <table
          className="table table-bordered table-condensed"
        >
          <tbody>
            <tr>
              <td>
                Key:
              </td>
              <td>
                <span
                  className="waiting glyphicon glyphicon-glass"
                />
                : 
                waiting
              </td>
              <td>
                <span
                  className="queued glyphicon glyphicon-calendar"
                />
                : 
                queued
              </td>
              <td>
                <span
                  className="running glyphicon glyphicon-console"
                />
                : 
                running
              </td>
              <td>
                <span
                  className="succeeded glyphicon glyphicon-ok"
                />
                : 
                succeeded
              </td>
              <td>
                <span
                  className="failed glyphicon glyphicon-remove"
                />
                : 
                failed
              </td>
              <td>
                <span
                  className="upstream_failed glyphicon glyphicon-minus"
                />
                : 
                upstream_failed
              </td>
            </tr>
          </tbody>
        </table>
        <button
          className="btn btn-default"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Start new run
        </button>
      </div>
    </div>
  </div>
</div>
`;
